actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1264,8 +1278,16 @@ def resources(*resources, &block)
             concerns(options[:concerns]) if options[:concerns]
 
             collection do
-              get  :index if parent_resource.actions.include?(:index)
-              post :create if parent_resource.actions.include?(:create)
+              actions = parent_resource.actions
+              (options[:constraints] ||= {}).merge! parent_resource.collection_param => COLLECTION_PARAM_REGEXP if parent_resource.collection_routing?
+              get    :index, options if actions.include?(:index)
+              post   :create, options if actions.include?(:create)
+              if parent_resource.collection_routing?
+                put    :replace, options if actions.include?(:replace)
+                patch  :update_many, options if actions.include?(:update_many)
+                delete :destroy_many, options if actions.include?(:destroy_many)
+                get    :edit_many, options.merge(as: 'edit') if actions.include?(:edit_many)
+              end
             end
 
             new do
```

Comment:
Because of the change above, I think you have to move this `new` block **above** the `collection` block.

Since routes are evaluated in order, if you don't do that, a route like `/posts/new` will be processed like an index action where "new" is the parameter:

```sh
Started GET "/posts/new" for ::1 at 2015-01-12 12:00:14 -0800
Processing by PostsController#index as HTML
  Parameters: {"collection"=>true, "ids"=>"new"}
```

If you move the `new` block above the `collection`, then you get the expected result:

```sh
Started GET "/posts/new" for ::1 at 2015-01-12 12:05:32 -0800
Processing by PostsController#new as HTML
  Rendered posts/_form.html.erb (5.1ms)
  Rendered posts/new.html.erb within layouts/application (8.5ms)
```